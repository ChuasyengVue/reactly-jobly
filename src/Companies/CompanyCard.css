.link{
    text-decoration: none;
    color: inherit;
}

.link:hover{
    color: rgb(42, 42, 221);
}

.card-container{
    margin: 20px;
    background-color: white;
    
}

.company-logo{
    max-width: 50px;
    height: auto;
}

