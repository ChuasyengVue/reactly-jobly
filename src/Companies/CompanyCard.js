import React from "react";
import { Link } from 'react-router-dom';
import { Card } from 'reactstrap';
import "./CompanyCard.css";


const CompanyCard = ({ handle, name, description, logoUrl }) => {
    
    return(
        <div className="card-container">
            <Card>
                <Link to={`/companies/${handle}`} className='link'>
                    <h5> { name } </h5>
                    <p> { description } </p>
                    {logoUrl && <img src={ logoUrl } alt={ name } />}
                </Link>
            </Card>
        </div>
    )
}

export default CompanyCard;