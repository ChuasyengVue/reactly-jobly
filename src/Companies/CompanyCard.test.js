import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

it("matches snapshot with logo", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <CompanyCard
            handle="SpringBoard"
            name="SpringBoard"
            description="Become an exceptional developer in 16 weeks."
            logo_url=""
        />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot without logo", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <CompanyCard
            handle="SpringBoard"
            name="SpringBoard"
            description="Become a coder in 10months"
        />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
