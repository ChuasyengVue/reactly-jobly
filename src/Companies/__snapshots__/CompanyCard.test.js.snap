// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <div
    class="card-container"
  >
    <div
      class="card"
    >
      <a
        class="link"
        href="/companies/SpringBoard"
      >
        <h5>
           SpringBoard 
        </h5>
        <p>
           Become an exceptional developer in 16 weeks. 
        </p>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <div
    class="card-container"
  >
    <div
      class="card"
    >
      <a
        class="link"
        href="/companies/SpringBoard"
      >
        <h5>
           SpringBoard 
        </h5>
        <p>
           Become a coder in 10months 
        </p>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
